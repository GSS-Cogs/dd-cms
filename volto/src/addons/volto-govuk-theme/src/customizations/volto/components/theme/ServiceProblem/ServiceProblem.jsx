/**
 * Home container.
 * @module components/theme/ServiceProblem/ServiceProblem
 */

import React from 'react';
import { withServerErrorCode } from '@plone/volto/helpers/Utils/Utils';

/**
 * Service Problem function.
 * @function ServiceProblem
 * @returns {string} Markup of the not found page.
 */
const ServiceProblem = () => (
  <div class="app-width-container">
    <main
      class="govuk-main-wrapper govuk-main-wrapper--l"
      id="main-content"
      role="main"
    >
      <div class="govuk-grid-row">
        <div class="govuk-grid-column-two-thirds">
          <h1 class="govuk-heading-l">
            Sorry, there is a problem with the service
          </h1>
          <p class="govuk-body">
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
            eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim
            ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut
            aliquip ex ea commodo consequat.
          </p>
          <p class="govuk-body">Try again later.</p>
        </div>
      </div>
    </main>
  </div>
);

export default withServerErrorCode(500)(ServiceProblem);
