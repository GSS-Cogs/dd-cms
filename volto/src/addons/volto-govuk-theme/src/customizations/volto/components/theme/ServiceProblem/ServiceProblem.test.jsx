import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-intl-redux';
import { MemoryRouter } from 'react-router-dom';

import ServiceProblem from './ServiceProblem';

global.__SERVER__ = true; // eslint-disable-line no-underscore-dangle

const mockStore = configureStore();

describe('ServiceProblem', () => {
  it('renders a not found component', () => {
    const store = mockStore({
      intl: {
        locale: 'en',
        messages: {},
      },
    });
    const component = renderer.create(
      <Provider store={store}>
        <MemoryRouter>
          <ServiceProblem />
        </MemoryRouter>
      </Provider>,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
