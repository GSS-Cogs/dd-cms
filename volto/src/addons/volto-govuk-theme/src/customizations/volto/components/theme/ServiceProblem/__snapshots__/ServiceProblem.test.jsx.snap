// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServiceProblem renders a not found component 1`] = `
<div
  class="app-width-container"
>
  <main
    class="govuk-main-wrapper govuk-main-wrapper--l"
    id="main-content"
    role="main"
  >
    <div
      class="govuk-grid-row"
    >
      <div
        class="govuk-grid-column-two-thirds"
      >
        <h1
          class="govuk-heading-xl govuk-!-margin-bottom-9"
        >
          Sorry, there is a problem with the service
        </h1>
        <p
          class="govuk-body"
        >
          Our Support Team is working on the error, and will resolve it as soon as possible.
        </p>
        <p
          class="govuk-body"
        >
          Try again later.
        </p>
      </div>
    </div>
  </main>
</div>
`;
